File: Cap 07/7.2 - Classificacao de Elementos/Numerico/clasnum.scala
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 7 - Topico 7.2
 *
 * Programa ..: CLASSIFICACAO DE LISTA POR NUMERO
 */

object Main {
  def main(args: Array[String]): Unit = {
  
    var NUMEROS: Array[Int] = new Array[Int](5)
    var I, J, X: Int = 0

    // Trecho de entrada de dados
    
    println("CLASSIFICACAO DE NUMEROS (CRESCENTE)\n")
    for (I <- 0 to 4 by 1) {
      print(s"Entre o ${I + 1}o. numero: ")
      NUMEROS(I) = scala.io.StdIn.readInt()
    }

    // Trecho de processamento da classificacao
    
    for (I <- 0 to 3 by 1) {
      for (J <- I + 1 to 4 by 1) {
        if (NUMEROS(I) > NUMEROS(J)) {
          X = NUMEROS(I)
          NUMEROS(I) = NUMEROS(J)
          NUMEROS(J) = X
        }
      }
    }

    // Trecho de saida com dados classificados
    
    println("\nNUMEROS CLASSIFICADOS\n")
    for (I <- 0 to 4 by 1) {
      println(NUMEROS(I))
    }
    
  }
}

File: Cap 07/7.2 - Classificacao de Elementos/Numerico/clasnum.go
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 7 - Topico 7.2
 * 
 * Programa ..: LISTA CLASSIFICADA POR NUMERO
 */

package main

import (
  "bufio"
  "fmt"
  "os"
  "strconv"
)

var I, J, X int
func main() {
  
  NUMEROS := make([]int, 5) 
  
  // Trecho de entrada de dados

  fmt.Println("CLASSIFICACAO DE NUMEROS (CRESCENTE)\n")
  leia := bufio.NewReader(os.Stdin)
  for I = 0; I <= 4; I++ {
    fmt.Printf("Entre o %do. numero: ", I + 1)
    entrada, _ := leia.ReadString('\n')
    numero, _ := strconv.Atoi(entrada[:len(entrada)-1])
    NUMEROS[I] = numero
  }

  // Trecho de processamento da classificacao

  for I = 0; I <= 3; I++ {
    for J = I + 1; J <= 4; J++ {
      if (NUMEROS[I] > NUMEROS[J]) {
        X = NUMEROS[I]
        NUMEROS[I] = NUMEROS[J]
        NUMEROS[J] = X
      }
    }
  }

  // Trecho de saida com dados classificados

  fmt.Println("\nNUMEROS CLASSIFICADOS\n")
  for I = 0; I <= 4; I++ {
    fmt.Println(NUMEROS[I])
  }
}
